### ISSUE TITLE
Java2dProcessor fails at reading EXIF data

### ISSUE THREAD

**Issue by kahlep** [cantaloupe-project/cantaloupe#454]
Dear Cantaloupe team,

we are facing an issue with a set of jpg files with possibly corrupt EXIF data that Cantaloupe 4.1.7 (running in Tomcat 8.5) is not able to serve.

The application is configured as a proxy to our image server as `HttpSource`. Images need to be rotated according to their EXIF orientation, which is why we chose ImageIO and GraphicsMagick as fallback:

```
processor.ManualSelectionStrategy.jpg = Java2dProcessor
processor.ManualSelectionStrategy.fallback = GraphicsMagickProcessor
...
processor.metadata.respect_orientation = true
```

On a GET request to the info.json of an affected file, I would expect the application to load the image and its metadata using ImageIO and, in case of failure, to do the same again using GraphicsMagick. 

But the ImageIO plugin fails with a `ClassCastException` when reading the EXIF data.
Instead of choosing the fallback, Cantaloupe responds with status 500 and the following content (the application log does not show that stacktrace):

```
500 Internal Server Error

[B cannot be cast to [Ljava.lang.String;

java.lang.ClassCastException: [B cannot be cast to [Ljava.lang.String;
	at it.geosolutions.imageio.plugins.tiff.TIFFField.getAsString(TIFFField.java:1164)
	at it.geosolutions.imageioimpl.plugins.tiff.TIFFImageMetadata.getStandardTextNode(TIFFImageMetadata.java:755)
	at javax.imageio.metadata.IIOMetadata.getStandardTree(IIOMetadata.java:723)
	at it.geosolutions.imageioimpl.plugins.tiff.TIFFImageMetadata.getAsTree(TIFFImageMetadata.java:218)
	at com.sun.media.imageio.plugins.tiff.TIFFDirectory.createFromMetadata(TIFFDirectory.java:224)
	at edu.illinois.library.cantaloupe.processor.codec.Util.readOrientation(Util.java:145)
	at edu.illinois.library.cantaloupe.processor.codec.JPEGMetadata.getExifOrientation(JPEGMetadata.java:64)
	at edu.illinois.library.cantaloupe.processor.codec.JPEGMetadata.getOrientation(JPEGMetadata.java:97)
	at edu.illinois.library.cantaloupe.processor.AbstractImageIOProcessor.getEffectiveOrientation(AbstractImageIOProcessor.java:109)
	at edu.illinois.library.cantaloupe.processor.AbstractImageIOProcessor.readInfo(AbstractImageIOProcessor.java:79)
	at edu.illinois.library.cantaloupe.cache.InfoService.readInfo(InfoService.java:198)
	at edu.illinois.library.cantaloupe.cache.InfoService.getOrReadInfo(InfoService.java:134)
	at edu.illinois.library.cantaloupe.cache.CacheFacade.getOrReadInfo(CacheFacade.java:67)
	at edu.illinois.library.cantaloupe.resource.PublicResource.getOrReadInfo(PublicResource.java:110)
	at edu.illinois.library.cantaloupe.resource.iiif.v2.InformationResource.doGET(InformationResource.java:143)
	at edu.illinois.library.cantaloupe.resource.HandlerServlet.handle(HandlerServlet.java:98)
	at edu.illinois.library.cantaloupe.resource.HandlerServlet.doGet(HandlerServlet.java:36)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:635)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
	at psiprobe.Tomcat85AgentValve.invoke(Tomcat85AgentValve.java:35)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
	at org.apache.coyote.ajp.AjpProcessor.service(AjpProcessor.java:479)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:800)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1471)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
```

Besides stripping the EXIF data from the affected files, is there any way to circumvent this issue?

Please find a crop of one of the files [here](https://user-images.githubusercontent.com/3034916/111152798-6ff4d300-8591-11eb-9048-4dce805d8e21.jpg)

Thanks in advance and best wishes

**Comment by adolski** [comment]
Hi @kahlep ,

The ClassCastException is caused by a bug. I've pushed a fix for that (above) that will appear in 4.1.8.

The "fallback processor" is perhaps mis-named--it's really just the default processor that gets used when there is no other format match. It's not used as a fallback in the event of an error..

**Comment by kahlep** [comment]
Hi @adolski,
thank you very much for the quick fix (and clarification)! Works like a charm now.
